### ISSUE TITLE
Generate random affine

### ISSUE THREAD

**Issue by skoudoro** [dipy/dipy#1687]
## Description

As suggested on [this comment](https://github.com/nipy/dipy/pull/1684#issuecomment-445891112), it would be good to have a function for generating random affine matrix. 
This is just a reminder.

**Comment by jhlegarreta** [comment]
@skoudoro Is this still relevant/useful? The proposed method looks straightforward to implement. Should we go for it, or else, should we control the linear and affine (dropping the projective part?) cases separately, like in [here](https://stackoverflow.com/questions/16682965/how-to-generaterandomtransform-with-opencv)?

**Comment by skoudoro** [comment]
Yes @jhlegarreta, still relevant and you can go for it. This function will be really useful for our tests like registration module.

**Comment by sidkapoor97** [comment]
@arokem @skoudoro is there any link which helps in testing whether a unit determinant matrix represents affine transformation or not? Also rotation and translation have determinant 1, whereas shear doesn't.

**Comment by skoudoro** [comment]
close based on this comment: https://github.com/dipy/dipy/pull/1709#issuecomment-5732504140
